Checkout/Checkout/Source/Validation/Validators/CardValidator.swift

```diff
@@ -178,63 +203,69 @@ public class CardValidator: CardValidating {
   // MARK: - Private
 
   private func validateExpiry(
-    expiryMonth: String,
-    expiryYear: String
+    expiryMonth: Int,
+    expiryYear: Int
   ) -> Result<ExpiryDate, ValidationError.ExpiryDate> {
-    guard let month = Int(expiryMonth) else {
-      return .failure(.invalidMonthString)
-    }
-
-    guard let year = Int(expiryYear) else {
-      return .failure(.invalidYearString)
-    }
-
-    guard month >= 1 && month <= 12 else {
+    guard expiryMonth >= 1 && expiryMonth <= 12 else {
       return .failure(.invalidMonth)
     }
 
-    let fourDigitYear: Int
-
-    switch self.fourDigitYear(
-      from: expiryYear,
-      year: year) {
-    case .success(let fourDigitYearResult):
-      fourDigitYear = fourDigitYearResult
-    case .failure(let expiryDateError):
-      return .failure(expiryDateError)
-    }
-
     let currentDate = Date()
+    // using UTC-12 as this is the latest timezone where a card could be valid
+    let cardExpiryComponents = DateComponents(
+      timeZone: .utcMinus12,
+      year: expiryYear,
+      month: expiryMonth
+    )
 
-    guard let providedDate = calendar.date(
-      from: DateComponents(
-        year: fourDigitYear,
-        month: month
-      )
-    ), providedDate >= currentDate else {
+    guard
+      let cardExpiryDate = calendar.date(from: cardExpiryComponents),
+      let nextMonth = calendar.date(byAdding: .month, value: 1, to: cardExpiryDate),
+      nextMonth > currentDate
+    else {
       return .failure(.inThePast)
     }
 
     let expiryDate = ExpiryDate(
-      month: month,
-      year: fourDigitYear)
+      month: expiryMonth,
+      year: expiryYear)
 
     return .success(expiryDate)
   }
 
-  private func fourDigitYear(
-    from expiryYear: String,
-    year: Int
-  ) -> Result<Int, ValidationError.ExpiryDate> {
+  private func fourDigitYear(from expiryYear: String) -> Result<Int, ValidationError.ExpiryDate> {
     switch expiryYear.count {
-    case 0, 1, 3:
-      return .failure(.incompleteYear)
-    case 2:
-      return .success(2000 + year)
-    case 4:
-      return .success(year)
+    case 0...4:
+      guard let year = Int(expiryYear) else {
+        return .failure(.invalidYearString)
+      }
+
+      return fourDigitYear(from: year)
     default:
       return .failure(.invalidYear)
     }
   }
+
+  private func fourDigitYear(from expiryYear: Int) -> Result<Int, ValidationError.ExpiryDate> {
+    // year must be 2 or 4+ digits, otherwise year is incomplete
+    // 0 digits - no year has been entered yet
+    // 1 or 3 digits - we are waiting on an extra digit
+    guard expiryYear > 1000 || (expiryYear > 9 && expiryYear < 100) else {
```

Comment:
please add example for this `expiryYear > 9 && expiryYear < 100` and why starts AFTER `9`?